contracts/core/BaseAccount.sol

```diff
@@ -31,6 +41,64 @@ abstract contract BaseAccount is IAccount {
      */
     function entryPoint() public view virtual returns (IEntryPoint);
 
+    /**
+     * execute a single call from the account.
+     */
+    function execute(address target, uint256 value, bytes calldata data) virtual external {
+        _requireForExecute();
+
+        bool ok = Exec.call(target, value, data, gasleft());
+        if (!ok) {
+            Exec.revertWithReturnData();
+        }
+    }
+
+    /**
+     * execute a batch of calls.
+     * revert on the first call that fails.
+     * If the batch reverts, and it contains more than a single call, then wrap the revert with ExecuteError,
+     *  to mark the failing call index.
+     */
+    function executeBatch(Call[] calldata calls) external {
+        _requireForExecute();
+
+        for (uint256 i = 0; i < calls.length; i++) {
+            Call calldata call = calls[i];
+            bool ok = Exec.call(call.target, call.value, call.data, gasleft());
+            if (!ok) {
+                if (calls.length == 1) {
+                    Exec.revertWithReturnData();
+                } else {
+                    revert ExecuteError(i, Exec.getReturnData(0));
+                }
+
+            }
+        }
+    }
+
+    /**
+     * check current account deposit in the entryPoint
+     */
+    function getDeposit() public view returns (uint256) {
+        return entryPoint().balanceOf(address(this));
+    }
+
+    /**
+     * deposit more funds for this account in the entryPoint
+     */
+    function addDeposit() public payable {
+        entryPoint().depositTo{value: msg.value}(address(this));
+    }
+
+    /**
+     * withdraw value from the account's deposit
+     * @param withdrawAddress target to send to
+     * @param amount to withdraw
+     */
+    function withdrawDepositTo(address payable withdrawAddress, uint256 amount) public onlyOwner {
+        entryPoint().withdrawTo(withdrawAddress, amount);
+    }
```

Comment:
I am not convinced these three must be a part of the BaseAccount. This are utility functions that we can trust the wallets to implement themselves.

Reply:
we now have 2 wallets (SimpleAccount, Simple7702Account) and both ned these methods.
A deposit is an extension of account's balance, and without it, it can't easily use it.
(It can call the entrypoint - but that requires the application to know the entrypoint interface)

Reply:
actually, I agree: Simple7702Account doesn't need a deposit: it can be used as a direct batch call if it doesn't use a paymaster. removing from base.